tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include "testing_common_defs.h"
+#include <addr.h>
+#include <keys.h>
+#include "mocks/conn.h"
+#include "mocks/hci_core.h"
+#include "mocks/keys_help_utils.h"
+#include "mocks/hci_core_expects.h"
+
+/* This LUT contains different combinations of ID and Address pairs */
+extern const struct id_addr_pair testing_id_addr_pair_lut[CONFIG_BT_MAX_PAIRED];
+
+/* This list holds returned references while filling keys pool */
+extern struct bt_keys *returned_keys_refs[CONFIG_BT_MAX_PAIRED];
+
+/* Pointer to the current set of oldest testing parameter */
+struct id_addr_pair const *oldest_params_vector;
+
+/* This call back will be called when bt_unpair() is called with the same parameters
+ * It will clear the key slot with the assigned to the 'id'
+ */
+static int bt_unpair_custom_fake(uint8_t id, const bt_addr_le_t *addr)
+{
+	struct bt_keys *keys = NULL;
+
+	keys = bt_keys_find_addr(id, addr);
+	if (keys) {
+		bt_keys_clear(keys);
+	}
+
+	/* This check is used here because bt_unpair() is called with a local variable address */
+	expect_single_call_bt_unpair(oldest_params_vector->id, oldest_params_vector->addr);
+
+	return 0;
+}
+
+/* This call back will be called when bt_unpair() is called with the same parameters
+ * It can be used when a call to bt_unpair() isn't expected
+ */
+static int bt_unpair_unreachable_custom_fake(uint8_t id, const bt_addr_le_t *addr)
+{
+	ARG_UNUSED(id);
+	ARG_UNUSED(addr);
+	zassert_unreachable("Unexpected call to 'bt_unpair()' occurred");
+	return 0;
+}
+
+/* This call back will be called when bt_conn_foreach() is called with the same parameters
+ * It will act as if there are some active connections
+ */
+static void bt_conn_foreach_custom_fake_key_slot_0_in_use(int type, bt_conn_foreach_cb func,
```

Comment:
Please either pick the end or the start for your `custom_fake` pre/suf-fix, but not the middle like this.
I would be in favor of removing it entirely, or just having a `fake_` prefix.